subworkflows/nf-core/fastq_align_dedup_bismark/main.nf

```diff
@@ -0,0 +1,154 @@
+     */
+    BISMARK_ALIGN (
+        ch_reads,
+        ch_fasta,
+        ch_bismark_index
+    )
+    ch_versions = ch_versions.mix(BISMARK_ALIGN.out.versions)
+
+    if (skip_deduplication) {
+    alignments        = BISMARK_ALIGN.out.bam
+    alignment_reports = BISMARK_ALIGN.out.report.map{ meta, report -> [ meta, report, [] ] }
+    } else {
+    /*
+    * Run deduplicate_bismark
+    */
+    BISMARK_DEDUPLICATE (
+        BISMARK_ALIGN.out.bam
+    )
+    alignments        = BISMARK_DEDUPLICATE.out.bam
+    alignment_reports = BISMARK_ALIGN.out.report.join(BISMARK_DEDUPLICATE.out.report)
+    ch_versions       = ch_versions.mix(BISMARK_DEDUPLICATE.out.versions)
+    }
+
+    /*
+     * Run bismark_methylation_extractor
+     */
+    BISMARK_METHYLATIONEXTRACTOR (
+        alignments,
+        ch_bismark_index
+    )
+    ch_methylation_bedgraph = BISMARK_METHYLATIONEXTRACTOR.out.bedgraph
+    ch_methylation_calls    = BISMARK_METHYLATIONEXTRACTOR.out.methylation_calls
+    ch_methylation_coverage = BISMARK_METHYLATIONEXTRACTOR.out.coverage
+    ch_methylation_report   = BISMARK_METHYLATIONEXTRACTOR.out.report
+    ch_methylation_mbias    = BISMARK_METHYLATIONEXTRACTOR.out.mbias
+    ch_versions             = ch_versions.mix(BISMARK_METHYLATIONEXTRACTOR.out.versions)
+
+    /*
+     * Run coverage2cytosine
+     */
+    if (cytosine_report) {
+    BISMARK_COVERAGE2CYTOSINE (
+        BISMARK_METHYLATIONEXTRACTOR.out.coverage,
+        ch_fasta,
+        ch_bismark_index
+    )
+    ch_coverage2cytosine_coverage = BISMARK_COVERAGE2CYTOSINE.out.coverage.collect{ it[1] }
+    ch_coverage2cytosine_report   = BISMARK_COVERAGE2CYTOSINE.out.report.collect{ it[1] }
+    ch_coverage2cytosine_summary  = BISMARK_COVERAGE2CYTOSINE.out.summary.collect{ it[1] }
+    ch_versions                   = ch_versions.mix(BISMARK_COVERAGE2CYTOSINE.out.versions)
+    }
+
+    /*
+     * Generate bismark sample reports
+     */
+    BISMARK_REPORT (
+        alignment_reports
+            .join(BISMARK_METHYLATIONEXTRACTOR.out.report)
+            .join(BISMARK_METHYLATIONEXTRACTOR.out.mbias)
+    )
+    ch_bismark_report = BISMARK_REPORT.out.report
+    ch_versions       = ch_versions.mix(BISMARK_REPORT.out.versions)
+
+    /*
+     * Generate bismark summary report
+     */
+    BISMARK_SUMMARY (
+        BISMARK_ALIGN.out.bam.collect{ it[1].name }.ifEmpty([]),
+        alignment_reports.collect{ it[1] }.ifEmpty([]),
+        alignment_reports.collect{ it[2] }.ifEmpty([]),
+        BISMARK_METHYLATIONEXTRACTOR.out.report.collect{ it[1] }.ifEmpty([]),
+        BISMARK_METHYLATIONEXTRACTOR.out.mbias.collect{ it[1] }.ifEmpty([])
+    )
+    ch_bismark_summary = BISMARK_SUMMARY.out.summary
+    ch_versions        = ch_versions.mix(BISMARK_SUMMARY.out.versions)
+
+    /*
+     * MODULE: Run samtools sort on dedup bam
+     */
+    SAMTOOLS_SORT_DEDUPLICATED (
+        alignments,
+        [[:],[]] // Empty map and list as is optional input but required for nextflow
```

Comment:
My preference is usually to describe what the input is in the comment since we can already see it's an empty input.